.github/workflows/release.yml

```diff
@@ -0,0 +1,267 @@
+# while the release workflow runs. In case the git history would not be linear, this workflow will
+# fail without having uploaded/pushed any release artifacts.
+
+
+name: Release
+
+on:
+  # Manually triggered
+  workflow_dispatch:
+    inputs:
+      releaseVersion:
+        description: 'The version to release - e.g. `0.5.0`'
+        required: true
+      nextVersion:
+        description: 'The nest development version - e.g. `0.6.0` - without the SNAPSHOT suffix'
+        required: true
+
+jobs:
+  build:
+    name: Built, test, release
+    runs-on: ubuntu-latest
+
+    # Restrict "Release" to these people
+    # (maybe there's a way via the GH API to _correctly_ identify whether the actor is an admin,
+    # owner or moderator, but that either requires using a 3rd party action or a bunch of code.)
+    if: contains('["rymurr", "laurentgo", "snazy"]', github.actor)
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+
+    # Log changelog (FYI only)
+    - name: Changelog for ${{ github.event.inputs.releaseVersion }}
+      env:
+        RELEASE_VERSION: ${{ github.event.inputs.releaseVersion }}
+      run: |
+        # git changelog -t ${RELEASE_VERSION} -n -x /dev/null | egrep -v '^  [*] (Bump |\[release\] ).*$'
+        git changelog -t ${RELEASE_VERSION} -n -x /dev/null
+
+    - name: Set up JDK 11
+      uses: actions/setup-java@v1
+      with:
+        java-version: 11
+        server-id: ossrh
+        server-username: MAVEN_USERNAME
+        server-password: MAVEN_OSSRH_TOKEN
+        gpg-private-key: ${{ secrets.MAVEN_GPG_PRIVATE_KEY }}
+        gpg-passphrase: MAVEN_GPG_PASSPHRASE
+
+    - name: Set up JDK 8
+      uses: AdoptOpenJDK/install-jdk@v1
+      with:
+        version: '8'
+        targets: 'JDK8_HOME'
+
+    # Using Python 3.8
+    # Might be worth to bump the version and/or make it configurable in the future
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.8
+
+    - name: Set up Maven toolchains.xml
+      run: |
+        cat << 'EOT' > ${HOME}/.m2/toolchains.xml
+        <?xml version="1.0" encoding="UTF8"?>
+        <toolchains>
+          <toolchain>
+            <type>jdk</type>
+            <provides>
+              <version>11</version>
+              <vendor>zulu</vendor>
+            </provides>
+            <configuration>
+              <jdkHome>${env.JAVA_HOME}</jdkHome>
+            </configuration>
+          </toolchain>
+          <toolchain>
+            <type>jdk</type>
+            <provides>
+              <version>1.8</version>
+              <vendor>adoptopenjdk</vendor>
+            </provides>
+            <configuration>
+              <jdkHome>${env.JDK8_HOME}</jdkHome>
+            </configuration>
+          </toolchain>
+        </toolchains>
+        EOT
+
+    # TODO Maybe better to _not_ use cached dependencies just to be sure that everything's available at time of release --> remove this one?
```

Comment:
I think its prob best to keep the caches. Dont feel strongly though :man_shrugging:

Reply:
Yup, I don't mind.